climada/engine/impact.py

```diff
@@ -1389,6 +1389,114 @@ def _selected_exposures_idx(self, coord_exp):
             LOGGER.warning("No exposure coordinates match the selection.")
         return sel_exp
 
+    @classmethod
+    def concat(cls, imp_list, reset_event_ids = False):
+        """
+        Concatenate impact objects with the same exposure.
+        This function is useful if, e.g. different impact functions
+        have to be applied for different seasons (e.g. for agricultural impacts).
+
+        It concatenates the following attributes:
+            event_id, event_name, date, frequency, imp_mat, at_event, eai_exp, aai_exp
+        and takes the following attributes from the first impact object in the passed imp_list
+            coord_exp, crs, unit, tot_value, tag
```

Comment:
Please add some explanations on how this works:
1. What happens if all impact matrices are empty?
2. Several values are *not* concatenated but *summed up*
3. What happens to the event IDs?